proposals/text/7462-rag-evaluation.md

```diff
@@ -0,0 +1,243 @@
+available metrics for all queries.
+
+
+```python
+def detailed_comparative_summary(self, other: "RAGPipelineEvaluation"):
+```
+
+```bash
+{
+    "queries": {
+      "query_id": ["53c3b3e6, 225f87f7, 8ac473ec, 97d284ca"],
+      "question": ["What is the capital of France?", "What is the capital of Spain?"],
+      "contexts": ["wiki_France", "wiki_Spain"],
+      "answer": ["Paris", "Madrid"]  
+    }
+    "pipeline_1": {  
+        "reciprocal_rank": [0.378064, 0.534964, 0.216058, 0.778642],
+        "single_hit": [1, 1, 0, 1],
+        "multi_hit": [0.706125, 0.454976, 0.445512, 0.250522],
+        "context_relevance": [0.805466, 0.410251, 0.750070, 0.361332],
+        "faithfulness": [0.135581, 0.695974, 0.749861, 0.041999],
+        "semantic_answer_similarity": [0.971241, 0.159320, 0.019722, 1],
+        "aggregated_score":
+          {
+              'Reciprocal Rank': 0.448,
+              'Single Hit': 0.5,
+              'Multi Hit': 0.540,
+              'Context Relevance': 0.537,
+              'Faithfulness': 0.452,
+              'Semantic Answer Similarity': 0.478
+          }
+    },
+    "pipeline_2": {
+        "reciprocal_rank": [0.378064, 0.534964, 0.216058, 0.778642],
+        "single_hit": [1, 1, 0, 1],
+        "multi_hit": [0.706125, 0.454976, 0.445512, 0.250522],
+        "context_relevance": [0.805466, 0.410251, 0.750070, 0.361332],
+        "faithfulness": [0.135581, 0.695974, 0.749861, 0.041999],
+        "semantic_answer_similarity": [0.971241, 0.159320, 0.019722, 1],
+        "aggregated_score":
+          {
+              'Reciprocal Rank': 0.448,
+              'Single Hit': 0.5,
+              'Multi Hit': 0.540,
+              'Context Relevance': 0.537,
+              'Faithfulness': 0.452,
+              'Semantic Answer Similarity': 0.478
+          }
+      }
+}
+```
+
+Have a method to find interesting scores thresholds, typically used for error analysis, for all metrics available.
+Some potentially interesting thresholds to find are: the 25th percentile, the 75th percentile, the mean , the median.
+
+```python
+def find_thresholds(self, metrics: List[str]) -> Dict[str, float]:
+```
+
+```bash
+data  = {
+    "thresholds": ["25th percentile", "75th percentile", "mean", "average"],
+    "reciprocal_rank": [0.378064, 0.534964, 0.216058, 0.778642],
+    "context_relevance": [0.805466, 0.410251, 0.750070, 0.361332],
+    "faithfulness": [0.135581, 0.695974, 0.749861, 0.041999],
+    "semantic_answer_similarity": [0.971241, 0.159320, 0.019722, 1],
+}
+````
+
+Then have another method that  
+
+```python
+def get_scores_below_threshold(self, metric: str, threshold: float):
+    """Get the all the queries with a score below a certain threshold for a given metric"""  
+```
+
+
+# Drawbacks
+
+- Relying on pandas DataFrame internally makes it easy to perform many of the operations.
+- Nevertheless, it can be burden, since we are making `pandas` a dependency of `haystack-ai`.
```

Comment:
It already is because of the Document class but we indeed don't strengthen this dependency.